src/test/java/net/testudobank/tests/MvcControllerIntegTest.java

```diff
@@ -239,6 +239,69 @@ public void testWithdrawTriggersOverdraft() throws SQLException, ScriptException
+   * @throws ScriptException
+   */
+
+  @Test
+  public void testWithdrawOverdraftOverage() throws SQLException, ScriptException { 
+
+   //initialize customer1 with a balance of $100. this will be represented as pennies in DB.
+   double CUSTOMER1_BALANCE = 100;
+   int CUSTOMER1_BALANCE_IN_PENNIES = MvcControllerIntegTestHelpers.convertDollarsToPennies(CUSTOMER1_BALANCE);
+   MvcControllerIntegTestHelpers.addCustomerToDB(dbDelegate, CUSTOMER1_ID, CUSTOMER1_PASSWORD, CUSTOMER1_FIRST_NAME, CUSTOMER1_LAST_NAME, CUSTOMER1_BALANCE_IN_PENNIES);
+
+   //Prepare Withdraw Form to withdraw $1099 from this customer's account.
+   double CUSTOMER1_AMOUNT_TO_WITHDRAW = 1099; 
+   User customer1WithdrawFormInputs = new User();
+   customer1WithdrawFormInputs.setUsername(CUSTOMER1_ID);
+   customer1WithdrawFormInputs.setPassword(CUSTOMER1_PASSWORD);
+   customer1WithdrawFormInputs.setAmountToWithdraw(CUSTOMER1_AMOUNT_TO_WITHDRAW);
+
+   //Store the timestamp of the withdraw request to verify it in the TransactionHistory table later
+   LocalDateTime timeWhenWithdrawRequestSent = MvcControllerIntegTestHelpers.fetchCurrentTimeAsLocalDateTimeNoMilliseconds();
+   System.out.println("Timestamp when withdraw request sent: " + timeWhenWithdrawRequestSent);
+
+
+   //verify that customer1's Overdraft balance is equal to remaining withdraw amount with interest applied
+   int CUSTOMER1_ORIGINAL_BALANCE_IN_PENNIES = MvcControllerIntegTestHelpers.convertDollarsToPennies(CUSTOMER1_BALANCE);
+   int CUSTOMER1_AMOUNT_TO_WITHDRAW_IN_PENNIES = MvcControllerIntegTestHelpers.convertDollarsToPennies(CUSTOMER1_AMOUNT_TO_WITHDRAW);
+   int CUSTOMER1_EXPECTED_OVERDRAFT_BALANCE_BEFORE_INTEREST_IN_PENNIES = CUSTOMER1_AMOUNT_TO_WITHDRAW_IN_PENNIES - CUSTOMER1_ORIGINAL_BALANCE_IN_PENNIES;
+   int CUSTOMER1_EXPECTED_OVERDRAFT_BALANCE_AFTER_INTEREST_IN_PENNIES = (int)(CUSTOMER1_EXPECTED_OVERDRAFT_BALANCE_BEFORE_INTEREST_IN_PENNIES * MvcController.INTEREST_RATE);
+   System.out.println("Expected Overdraft Balance in pennies: " + CUSTOMER1_EXPECTED_OVERDRAFT_BALANCE_AFTER_INTEREST_IN_PENNIES);
+
+   
+   //Check the response when the withdraw request is submitted. This should return the user back to the home screen due to an invalid request
+   String responsePage = controller.submitWithdraw(customer1WithdrawFormInputs);
+   assertEquals("welcome", responsePage);
+
+   //Fetch customer1's data from DB
+   List<Map<String, Object>> customersTableData = jdbcTemplate.queryForList("SELECT * FROM Customers;");
+
+   //Since the request did not go through, the balance is supposed to stay the same.
+   Map<String, Object> customer1Data = customersTableData.get(0);
+   assertEquals(CUSTOMER1_BALANCE_IN_PENNIES, (int)customer1Data.get("Balance"));
+
+   //check that TransactionHistory table is empty
+   List<Map<String, Object>> transactionHistoryTableData = jdbcTemplate.queryForList("SELECT * FROM TransactionHistory;");
+   assertEquals(true, transactionHistoryTableData.isEmpty());
```

Comment:
Just use `assertTrue()` to make this easier to read.

Reply:
Will do so. I originally used assertEquals because that is what was imported at the top, but I added another import statement to allow assertTrue() as well.